### ISSUE TITLE
train_step method is not called on QAngaroo

### ISSUE THREAD

**Issue by james-assiene** [facebookresearch/ParlAI#1814]
**Bug description**
It seems like the "train_step" method of a ParlAI agent is not called when training on QAngaroo

**Reproduction steps**

Code of the "hello.py" (HelloAgent):

```
import torch

from parlai.core.torch_agent import TorchAgent, Output

class HelloAgent(TorchAgent):
  
    def __init__(self, opt, shared=None):
        super().__init__(opt, shared)
        
    def train_step(self, batch):
        
        print("train")
        
        return Output(["train"])
      
    def eval_step(self, batch):
        
        #print("eval")
        
        return Output(["eval"])
```

Cmd line used to train the model:

```
!python ~/ParlAI/parlai/scripts/train_model.py -m hello -t qangaroo -mf hello_model
```

You can find a simple colab notebook highlighting the problem here:
https://drive.google.com/open?id=1VczVfuKz1YbGskfqGA6VQ0Ld3B1OruAP

**Expected behavior**
The message "hello" should be printed during training

**Logs**

```
Dictionary: loading dictionary from hello_model.dict
[ num words =  339450 ]
[creating task(s): qangaroo]
loading:  /root/ParlAI/data/qangaroo/qangaroo_v1.1/wikihop/train.json
[ training... ]
[ time:2.0s total_exs:906 epochs:0.02 ] {'exs': 0, 'num_updates': 0}
[ time:4.0s total_exs:1781 epochs:0.04 ] {'exs': 0, 'num_updates': 0}
```

**Additional context**

However, the "eval_step" method is indeed called (it is sufficient to uncomment the "print("eval")" line to see it)

```
Dictionary: loading dictionary from hello_model.dict
[ num words =  339450 ]
[creating task(s): qangaroo]
loading:  /root/ParlAI/data/qangaroo/qangaroo_v1.1/wikihop/train.json
[ training... ]
eval
eval
eval
eval
eval
eval
eval
eval
```

**Comment by stephenroller** [comment]
Thanks for pointing this out. Looks like the problem is here:

https://github.com/facebookresearch/ParlAI/blob/e73ffd6cc5ac5b9e13f5011bf6af804d23691721/parlai/tasks/qangaroo/agents.py#L53

That should be “labels” not “label”.
